packages/contracts-bedrock/src/dispute/AnchorStateRegistry.sol

```diff
@@ -23,105 +24,192 @@ import { ISuperchainConfig } from "interfaces/L1/ISuperchainConfig.sol";
+        if (address(anchorGame) == address(0)) {
+            return (startingAnchorRoot.root, startingAnchorRoot.l2BlockNumber);
+        }
+
+        // Otherwise, return the anchor root.
+        return (Hash.wrap(anchorGame.rootClaim().raw()), anchorGame.l2BlockNumber());
+    }
+
+    /// @notice Determines whether a game is registered in the DisputeGameFactory.
+    /// @param _game The game to check.
+    /// @return Whether the game is factory registered.
+    function isGameRegistered(IDisputeGame _game) public view returns (bool) {
+        // Grab the game and game data.
+        (GameType gameType, Claim rootClaim, bytes memory extraData) = _game.gameData();
+
+        // Grab the verified address of the game based on the game data.
+        (IDisputeGame _factoryRegisteredGame,) =
+            disputeGameFactory.games({ _gameType: gameType, _rootClaim: rootClaim, _extraData: extraData });
+
+        // Return whether the game is factory registered.
+        return address(_factoryRegisteredGame) == address(_game);
+    }
+
+    /// @notice Determines whether a game is of a respected game type.
+    /// @param _game The game to check.
+    /// @return Whether the game is of a respected game type.
+    function isGameRespected(IDisputeGame _game) public view returns (bool) {
+        return _game.gameType().raw() == portal.respectedGameType().raw();
+    }
+
+    /// @notice Determines whether a game is blacklisted.
+    /// @param _game The game to check.
+    /// @return Whether the game is blacklisted.
+    function isGameBlacklisted(IDisputeGame _game) public view returns (bool) {
+        return portal.disputeGameBlacklist(_game);
+    }
+
+    /// @notice Determines whether a game is retired.
+    /// @param _game The game to check.
+    /// @return Whether the game is retired.
+    function isGameRetired(IDisputeGame _game) public view returns (bool) {
+        return _game.createdAt().raw() < portal.respectedGameTypeUpdatedAt();
+    }
+
+    /// @notice Determines whether a game resolved properly and the game was not subject to any
+    ///         invalidation conditions. The root claim of a proper game IS NOT guaranteed to be
+    ///         valid. The root claim of a proper game CAN BE incorrect and still be a proper game.
+    /// @param _game The game to check.
+    /// @return Whether the game is a proper game.
+    /// @return Reason why the game is not a proper game.
+    function isGameProper(IDisputeGame _game) public view returns (bool, string memory) {
+        if (!isGameRegistered(_game)) {
+            return (false, "game not factory registered");
+        }
+
+        // Must be respected game type.
+        if (!isGameRespected(_game)) {
+            return (false, "game type not respected");
+        }
+
+        // Must not be blacklisted.
+        if (isGameBlacklisted(_game)) {
+            return (false, "game blacklisted");
+        }
+
+        // Must be created after the gameRetirementTimestamp.
```

Comment:
`gameRetirementTimestamp` doesn't exist anywhere (cool name though.)

Is this supposed to be `respectedGameTypeUpdatedAt`?